### PR TITLE
Implemented CriterionVisitor for EnhancedSelection

### PR DESCRIPTION
**Pull request by MarioBlazek**

### CODE REVIEW

File: Core/Search/Solr/Query/CriterionVisitor/EnhancedSelection.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+class EnhancedSelection extends Field
+{
+    /**
+     * @inheritDoc
+     */
+    public function canVisit(Criterion $criterion)
+    {
+        return
+            $criterion instanceof EnhancedSelectionCriterion &&
```
Comment (emodric) [inline Core/Search/Solr/Query/CriterionVisitor/EnhancedSelection.php]: Isn't `instanceof` enough?

File: Core/Search/Solr/Query/CriterionVisitor/EnhancedSelection.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                $preparedValues = (array)$this->mapSearchFieldvalue($value, $fieldType);
+
+                foreach ($preparedValues as $prepValue) {
+                    $queries[] = $name . ':"' . $this->escapeQuote($this->toString($prepValue), true) . '"';
+                }
+            }
+        }
+
+        switch ($criterion->operator) {
+            case Operator::CONTAINS:
```
Comment (emodric) [inline Core/Search/Solr/Query/CriterionVisitor/EnhancedSelection.php]: `CONTAINS` ? Ehanced selection criterion doesn't support `Operator::CONTAINS`: https://github.com/netgen/NetgenEnhancedSelectionBundle/blob/master/API/Repository/Values/Content/Query/Criterion/EnhancedSelection.php#L21-L32

File: Core/Search/Solr/Query/CriterionVisitor/EnhancedSelection.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                $op = ' AND ';
+                break;
+            case Operator::IN:
+            case Operator::EQ:
```
Comment (emodric) [inline Core/Search/Solr/Query/CriterionVisitor/EnhancedSelection.php]: How can the operation for `Operator::EQ` be `OR` if `Operator::EQ` supports only a single string value and not an array?

**Review by emodric — CHANGES_REQUESTED** [review]

**Comment by MarioBlazek** [comment]
Pull request updated, @emodric please check.

**Review by emodric — APPROVED** [review]

**Pull request merged**